scapy/fields.py

```diff
@@ -1234,6 +1234,19 @@ def i2repr(self, pkt, x):
         return bytes_hex(x[:self.length_from(pkt)]).decode()
 
 
+class XLEStrLenField(XStrLenField):
+    def i2repr(self, pkt, x):
+        a = bytes_hex(x[:self.length_from(pkt)]).decode()
+        le = "".join(reversed([a[i:i + 2] for i in range(0, len(a), 2)]))
+        return hex(int("0x" + le, 16))
```

Comment:
Why reverse it again ?

Reply:
The idea is to have a long hexadecimal value that user can manipulate but automatically encode it as LE to send over the network. When the user wants to save a value b’\xaa\xbb’ it is stored as b’\xbb\xaa’ but when the packet is showed is presented b’\xaa\xbb’

Reply:
You should rather store it `b'\xaa\xbb`' and do the reverse part in `i2m` / `m2i`